src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,661 @@
+            generator,
+            output_prediction_format,
+            output_visualization_kwargs,
+        )
+
+        # 2. Prepare empty text conditioning. Model invocation: self.tokenizer, self.text_encoder
+        if self.empty_text_embedding is None:
+            self.encode_empty_text()
+
+        # 3. Preprocessing input image
+        image, padding, original_resolution = self.image_processor.preprocess(
+            image, processing_resolution, resample_method_input, check_input, device, dtype
+        )  # [N,3,PPH,PPW]
+
+        # 4. Encode input image into latent space. Model invocation: self.vae.encoder
+        image_latent, pred_latent = self.prepare_latent(
+            image, latents, generator, ensemble_size, batch_size
+        )  # [N*E,4,h,w], [N*E,4,h,w]
+
+        del image
+
+        batch_empty_text_embedding = self.empty_text_embedding.to(device=device, dtype=dtype).repeat(
+            batch_size, 1, 1
+        )  # [B,1024,2]
+
+        # 5. Denoising loop. Model invocation: self.unet
+        clean_latent = []
+
+        with self.progress_bar(total=num_images * ensemble_size * num_inference_steps) as progress_bar:
+            for i in range(0, num_images * ensemble_size, batch_size):
+                batch_image_latent = image_latent[i : i + batch_size]  # [B,4,h,w]
+                batch_pred_latent = pred_latent[i : i + batch_size]  # [B,4,h,w]
+                B = batch_image_latent.shape[0]
+
+                batch_text_embedding = batch_empty_text_embedding[:B]  # [B,2,1024]
+
+                self.scheduler.set_timesteps(num_inference_steps, device=device)
+
+                for t in self.scheduler.timesteps:
+                    batch_latent = torch.cat([batch_image_latent, batch_pred_latent], dim=1)  # [B,8,h,w]
+                    noise = self.unet(batch_latent, t, encoder_hidden_states=batch_text_embedding).sample  # [B,4,h,w]
+                    batch_pred_latent = self.scheduler.step(
+                        noise, t, batch_pred_latent, generator=generator
+                    ).prev_sample  # [B,4,h,w]
+                    progress_bar.update(B)
+
+                clean_latent.append(batch_pred_latent)
+
+                del batch_image_latent, batch_pred_latent, batch_text_embedding, batch_latent, noise
+
+        pred_latent = torch.cat(clean_latent, dim=0)  # [N*E,4,h,w]
+
+        del clean_latent, image_latent, batch_empty_text_embedding
+
+        # 6. Decode prediction from latent into pixel space. Model invocation: self.vae.decoder
+        prediction = torch.cat(
+            [
+                self.decode_prediction(pred_latent[i : i + batch_size])
+                for i in range(0, pred_latent.shape[0], batch_size)
+            ],
+            dim=0,
+        )  # [N*E,1,PPH,PPW]
+
+        if not output_latent:
+            pred_latent = None
+
+        # 7. Postprocess predictions
+        prediction = self.image_processor.unpad_image(prediction, padding)  # [N*E,1,PH,PW]
```

Comment:
any reason we call it `prediction` instead of `image` like we do with other pipelines?
no strong opinion though, just think it's easier to read if it we use same structure/variable name/method name as possible

Reply:
Yes, I answered it in another comment. The `.prediction` is never quite an image, as it does not have the semantics of an image; therefore, I thought that it might be misleading to call it that. E.g., depth prediction has 1 channel, normals 3, and other modalities can have other, arbitrary numbers of channels.

Reply:
ok!